AMM/project/contracts/exchange-contract/src/main.sw

```diff
@@ -77,8 +77,8 @@ impl Exchange for Contract {
         let mut added_liquidity = 0;
 
         // checking this because this will either result in a math error or adding no liquidity at all
-        require(asset_a_in_deposit != 0, TransactionError::DepositCannotBeZero);
-        require(asset_b_in_deposit != 0, TransactionError::DepositCannotBeZero);
+        require(asset_a_in_deposit != 0, TransactionError::ExpectedNonZeroDeposit);
```

Comment:
I'm wondering if we should distinguish between the assets so that the error will tell the user which asset is 0.

Reply:
Done.